src/rest/main/main.go

```diff
@@ -151,19 +155,7 @@ func prepareCACertificates() *x509.CertPool {
 // Returns corresponding tls.ClientAuthType value. Exits the process
 // if value is not valid ('none', 'cert' or 'auth')
 func getTLSClientAuthType() tls.ClientAuthType {
-	switch clientAuth {
-	case "none":
-		return tls.RequestClientCert
-	case "user":
-		return tls.RequestClientCert
-	case "cert":
-		if caFile == "" {
-			glog.Fatal("--cacert option is mandatory when --client_auth is 'cert'")
-		}
-		return tls.RequireAndVerifyClientCert
-	default:
-		glog.Fatalf("Invalid '--client_auth' value '%s'. "+
-			"Expecting one of 'none', 'cert' or 'user'", clientAuth)
-		return tls.RequireAndVerifyClientCert // dummy
-	}
+
+	return tls.RequestClientCert
+
```

Comment:
Are we supporting only one type of authentication?

Reply:
No, the authentication I added was ApiKey based, also BasicAuth is supported. I will add the client cert auth back as well, that was an oversight. I think I need to have seperate flags for each auth type, like -auth-key, -auth-basic, -auth-cert since before you could only enable one at a time. In our case, we want basic and ApiKey at the same time.